Gifski/Gifski.swift

```diff
@@ -4,53 +4,54 @@ import AVFoundation
 final class Gifski {
 	private(set) var isRunning = false
 	private var progress: Progress!
+    private var observation: NSKeyValueObservation?
 
 	// `progress.fractionCompleted` is KVO-compliant, but we expose this for convenience
-	var onProgress: ((_ progress: Double) -> Void)?
+	var onProgress: ((_ progress: Progress) -> Void)?
 
 	/**
 	- parameters:
 		- frameRate: Clamped to 5...30. Uses the frame rate of `inputUrl` if not specified.
 	*/
-	@discardableResult
 	func convertFile(
 		_ inputUrl: URL,
 		outputUrl: URL,
 		quality: Double = 1,
 		dimensions: CGSize? = nil,
 		frameRate: Int? = nil
-	) -> Progress {
+	) {
 		/// TODO: Find a better way to handle this
 		guard !isRunning else {
 			fatalError("Create a new instance if you want to run multiple conversions at once")
 		}
 
 		isRunning = true
 
-		progress = Progress(parent: nil, userInfo: [.fileURLKey: outputUrl])
-		progress.fileURL = outputUrl
+		progress = Progress(parent: Progress.current(), userInfo: [:])
```

Comment:
You don't need to specify the `userInfo: [:]` part.

Reply:
`[.fileURLKey: outputUrl]` was actually needed for the folder progress bar to work..